### PR TITLE
Generate consistent snapshot names using tf_label module

### PR DESCRIPTION
**Pull request by s2504s**
## What
* Generated consistent names using the `tf_label` module

## Why
* Make variables names uniform to support concurrent module instantiations

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -1,23 +1,16 @@
+module "tf_label" {
+  source = "github.com/cloudposse/tf_label"
+  namespace = "${var.namespace}"
+  stage = "${var.stage}"
+  name = "${var.name}"
+}
+
 resource "aws_ami_from_instance" "create_ami" {
-  name               = "${var.name}"
+  name               = "${module.tf_label.id}"
```
Comment (osterman) [inline main.tf]: Funky whitespace

Comment (s2504s) [reply main.tf]: Fixed. I've used `terraform fmt`

File: README.md [hunk 2]
```diff
@@ -17,10 +15,12 @@ resource "aws_instance" "web" {
 ```
 module "tf_ami_from_instance" {
```
Comment (const-bon) [inline README.md:19]: Let's use `terraform fmt` like spaces

**Comment by osterman** [comment]
Rebase

File: variables.tf [hunk 3]
```diff
@@ -1,5 +1,11 @@
-variable "name" { default = "terraform-ami" }
-variable "source_instance_id" {}
-variable "tags" {
-  type    = "map"
+variable "namespace" {
+  default = "test"
```
Comment (const-bon) [inline variables.tf]: Let's remove default value and leave only variable definition in module.
E.g. `variable "namespace" {}`

File: variables.tf [hunk 4]
```diff
@@ -1,5 +1,11 @@
...
+}
+
+variable "stage" {
+  default = "dev"
```
Comment (const-bon) [inline variables.tf]: Let's remove default value and leave only variable definition in module.
E.g. `variable "stage" {}`

File: main.tf [hunk 5]
```diff
@@ -1,10 +1,17 @@
+module "tf_label" {
+  source    = "github.com/cloudposse/tf_label"
```
Comment (osterman) [inline main.tf]: Use precise URL

File: README.md [hunk 6]
```diff
@@ -5,22 +5,25 @@
...
+  }
 }
 ```
 
 ### Create AMI from an AWS instance. Instance ID is required"
 ```
 module "tf_ami_from_instance" {
-  source  = "github.com/cloudposse/tf_ami_from_instance"
-  name = "ami-${md5(timestamp())}"
+  source             = "github.com/cloudposse/tf_ami_from_instance"
```
Comment (osterman) [inline README.md]: Use precise URL

File: main.tf [hunk 7]
```diff
@@ -1,10 +1,17 @@
...
+  namespace = "${var.namespace}"
+  stage     = "${var.stage}"
+  name      = "${var.name}"
+}
+
 resource "aws_ami_from_instance" "create_ami" {
```
Comment (osterman) [inline main.tf]: Resource name should just be `default`

File: main.tf (hunk 5 above)
Comment (const-bon) [inline main.tf]: Use full path `source    = "git::https://github.com/cloudposse/tf_label?ref=0.1.0"`

File: outputs.tf [hunk 8]
```diff
@@ -1,3 +1,3 @@
 output "ami_id" {
-  value = "${aws_ami_from_instance.create_ami.id}"
+  value = "${module.tf_label.id}"
```
Comment (osterman) [inline outputs.tf]: `aws_ami_from_instance.create_ami.id` is authoritative. Use that as the source of truth.

File: main.tf [hunk 9]
```diff
@@ -1,10 +1,17 @@
...
+  source    = "git::https://github.com/cloudposse/tf_label.git?ref=0.1.0"
+  namespace = "${var.namespace}"
+  stage     = "${var.stage}"
+  name      = "${var.name}"
+}
+
+resource "aws_ami_from_instance" "default" {
+  name                    = "${module.tf_label.id}"
+  source_instance_id      = "${var.source_instance_id}"
   snapshot_without_reboot = "true"
```
Comment (osterman) [inline main.tf]: Let's make this a parameter and default it to `true`.

**Review by osterman — APPROVED** [review]

**Pull request merged**